### PR TITLE
mlm_weights(), feat_spec(), flat_strip() if-else refactors

### PR DESCRIPTION
**Pull request by jeswan**
This PR refactors the following:
- adds getter for mlm_weights_dict to `JiantTransformersModel`
- add  getter for feat_spec to `JiantTransformersModel`
- simplifies `flat_strip()`

### CODE REVIEW

File: jiant/tasks/lib/templates/hacky_tokenization_matching.py [hunk 1]
```diff
@@ -31,72 +30,12 @@ def input_flat_strip(tokens):
...
-        return string
-
-
-def xlm_roberta_flat_strip(tokens, return_indices=False):
-    # TODO: Refactor to use general SentencePiece function  (issue #1181)
-    return albert_flat_strip(tokens=tokens, return_indices=return_indices)
-
-
-def albert_flat_strip(tokens, return_indices=False):
+def flat_strip(tokens, tokenizer, return_indices=False):
```
Comment (zphang) [inline jiant/tasks/lib/templates/hacky_tokenization_matching.py:33]: Do we have tests for this?

**Review by zphang — COMMENTED** [review]
Looks good overall, see comment about `flat_strip`

File: jiant/tasks/lib/templates/hacky_tokenization_matching.py (hunk 1 above)
Comment (jeswan) [reply jiant/tasks/lib/templates/hacky_tokenization_matching.py:33]: Not yet. I'll add some now

Comment (jeswan) [reply jiant/tasks/lib/templates/hacky_tokenization_matching.py:33]: We did not have tests before but I just added tests here: https://github.com/nyu-mll/jiant/pull/1289. I'll check that the added tests pass on this PR.

Comment (jeswan) [reply jiant/tasks/lib/templates/hacky_tokenization_matching.py:33]: Thanks for requesting the tests! Found a bug with my implementation.

**Pull request merged**